aspnetcore/fundamentals/target-aspnetcore.md

```diff
@@ -37,30 +37,102 @@ Referencing ASP.NET Core in this manner is only supported for projects targeting
+Blazor supports WebAssembly (WASM) and server-based [hosting models](xref:blazor/hosting-models). Unless there's a specific reason not to support both hosting models, a [Razor components](xref:blazor/components/index) library should support both hosting models. A Razor components library must use the [Microsoft.NET.Sdk.Razor SDK](xref:razor-pages/sdk).
 
 ### Support both hosting models
 
-To support Razor component consumption from both [Blazor Server](xref:blazor/hosting-models#blazor-server) and [Blazor WASM](xref:blazor/hosting-models#blazor-webassembly) projects, use the following instructions for your editor.
+To support Razor component consumption from both [Blazor Server](xref:blazor/hosting-models#blazor-server) and [Blazor WebAssembly](xref:blazor/hosting-models#blazor-webassembly) projects, use the following instructions for your editor.
 
 # [Visual Studio](#tab/visual-studio)
 
-Use the **Razor Class Library** project template. The template's **Support pages and views** checkbox should be deselected.
+Use the **Razor Class Library** project template. Disable the template's **Support pages and views** check box.
 
 # [Visual Studio Code](#tab/visual-studio-code)
 
 Run the following command in the integrated terminal:
 
 ```dotnetcli
-dotnet new razorclasslib
+dotnet new razorclasslib --support-pages-and-views false
 ```
 
 # [Visual Studio for Mac](#tab/visual-studio-mac)
 
-Use the **Razor Class Library** project template.
+Use the **Razor Class Library** project template. Disable the template's **Support pages and views** check box.
 
 ---
 
+::: moniker range=">= aspnetcore-5.0"
+
+The library generated from the project template:
+
+* Targets .NET 5 with the [Target Framework Moniker (TFM)](/dotnet/standard/frameworks) of `net5.0`.
+* Enables browser compatibilty checks for platform dependencies by including `browser` as a supported platform with the `SupportedPlatform` MSBuild item.
+* Adds a NuGet package reference for the [Microsoft.AspNetCore.Components.Web](https://www.nuget.org/packages/Microsoft.AspNetCore.Components.Web) package.
+
+Example:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk.Razor">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <SupportedPlatform Include="browser" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Components.Web" Version="5.0.0" />
+  </ItemGroup>
+
+</Project>
+```
+
+### Only support the Blazor Server hosting model
+
+Class libraries rarely only support Blazor Server apps. To only support Razor component consumption from [Blazor Server](xref:blazor/hosting-models#blazor-server) projects, target .NET 5.
```

Comment:
Perhaps something like, 

"If your class library requires Blazor server specific features such as access to CircuitHandlers or ProtectedBrowserStorage, or uses ASP.NET Core specific features such as middlewares, controllers, or Razor Pages in addition to Blazor components, then you would add a framework reference to ASP.NET Core"